### ISSUE TITLE
Parse errors are formatted differently, which trips up editors trying to parse them.

### ISSUE THREAD

**Issue by epost** [purescript/purescript#1098]
Emacs won't recognise parse errors correctly, meaning we don't get to jump to the error location and all the rest of it.

Compile error:
```
Error in module Database.AnyDB.Transaction:
Error in value declaration f:
Error at src/Database/AnyDB/Transaction.purs line 36, column 1 - line 43, column 1:
  Expression
    \x -> 3
  does not have type
    Prim.String
See https://github.com/purescript/purescript/wiki/Error-Code-ExprDoesNotHaveType for more information, or to contribute content related to this error.
```

Parse error:
```
"src/Database/AnyDB/Transaction.purs" (line 34, column 1):
unexpected RParen
expecting "module" or end of input
``` 

**Update:** It seems that deprecated feature errors are printed in yet another format:

    psc: It looks like you are trying to use a feature from a previous version of the compiler:
    Array notation is no longer supported. Use Array _ instead of [_].
    at "/Users/erik/dev/ex/examples-purescript/purescript-turtle/src/CanvasCompiler.purs" (line 27, column 1)

**Comment by hdgarrood** [comment]
Do you think it might be worth copying Elm and adding an option to report errors using JSON?

**Comment by paf31** [comment]
Yes, that's a neat idea.

**Comment by epost** [comment]
I didn't know that Elm did that, that would be great!

**Comment by hdgarrood** [comment]
Only since very recently: http://elm-lang.org/blog/compiler-errors-for-humans#towards-an-ide

**Comment by epost** [comment]
Interesting! I can never stop wondering whether some of these things (error reporting, code navigation and searching, etc) couldn't benefit from establishing some interop standards common to multiple languages. If such a thing existed, it should be a lot easier to create tooling for a language.

**Comment by epost** [comment]
Re error formatting: it seems that ghc, gcc and javac all use the same format for reporting errors, which is basically ``/path/to/File.purs:3:5:Some error message``. Emacs understand this out of the box, and it also colours the error according to the note/warning/error label that gcc additionally provides.

ghc:

    bla.hs:2:3:
        Not in scope: ‘putStrLnx’
        Perhaps you meant ‘putStrLn’ (imported from Prelude)
gcc:

    bla.c:2:3: warning: implicitly declaring library function 'printf' with type 'int (const char *, ...)'
          printf("hoi poes!\n");
          ^

javac:

    $ javac Bla.java && java -cp . Bla
    Bla.java:4: error: not a statement
            errortje
            ^

**Comment by epost** [comment]
Fixed in b8ed0305e267b54c0914a770bcedc89688a5d54a.
